### PR TITLE
Minor tasks changes and Docker sample for daemon

### LINKED ISSUE #25: Backport tasks modification done in offspot/metrics
This is a placeholder to not forget to back-port in this repo the relevant modifications that have been done in [offspot/metrics](https://github.com/offspot/metrics/pull/18) once we agree on the PR (e.g. the distinction between ruff_args and black_args, the fact that args cannot be None, ...)

### PR DESCRIPTION
**Pull request by benoit74**
Fix #25 

## Changes
- Add sample Docker test configuration for daemon processes
- Add support for coverage HTML report
- Simplify task arguments (no clear reason / documentation why `str | None` is needed)
- Fix few task arguments help

### CODE REVIEW

**Comment by benoit74** [comment]
Sorry for the force-push a bit late, but I realized that I made a mistake

File: .github/workflows/Tests.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ jobs:
           pip install -e .[test,scripts]
 
       - name: Run the tests
-        run: inv coverage --args "-vvv"
+        run: inv coverage --no-html --args "-vvv"
```
Comment (rgaudin) [inline .github/workflows/Tests.yaml:32]: Why is this necessary?

**Comment by rgaudin** [comment]
PR name is not very useful neither 😉

**Comment by benoit74** [comment]
I removed most changes, see the updated list of changes before reviewing.

I do not achieve to reproduce your issue regarding pytest escaping a non activated virtualenv, so I isolated this change in a specific issue: https://github.com/openzim/_python-bootstrap/issues/27

File: tasks.py [hunk 2]
```diff
@@ -8,95 +8,96 @@
...
-@task(optional=["args"], help={"args": "pytest additional arguments"})
-def coverage(ctx: Context, args: str | None = ""):
+@task(
+    optional=["args", "html"],
+    help={
+        "args": "pytest additional arguments",
+        "html": "flag to export html report",
+    },
+)
+def coverage(ctx: Context, args: str = "", *, html: bool = False):
```
Comment (rgaudin) [inline tasks.py:38]: With this the html option is not exposed to the hatch commands. I suggest we add an `html` one like

```toml
[tool.hatch.envs.test.scripts]
...
report-cov = "inv report-cov --html '{args}'"
html = "inv coverage --html --args '{args}'"
```

would be used as `hatch run test:html` which seems quite convenient

**Comment by benoit74** [comment]
I finally understood why we needed sometimes the `args = args or "."` thing.

It is needed like you said for hatch scripts which passes `--args '{args}'`, i.e. `--args ''` if no args are provided. We hence always get a string (no need for `str | None`, `str` is OK) but need the `args = args or "."` for tasks which expects a path (and not for the ones which are ok with an empty arg.

Please review again to confirm it is ok for you before merging, sorry for that late understanding, but at least it is now correctly typed + documented.

**Review by rgaudin — APPROVED** [review]
Works for me 👍
